nion/swift/ImageCanvasItem.py

```diff
@@ -255,34 +260,86 @@ def _repaint(self, drawing_context: DrawingContext.DrawingContext, canvas_bounds
+                baseline = canvas_bounds.height
+
+                # Background box
+                if self.__background:
+                    total_text_height = text_height * 2 + padding
+                    total_height = scale_marker_height + padding + total_text_height
+                    background_top = baseline - total_height
+
+                    if self.__draw_on_right:
+                        background_left = content_width - content_width  # 0
+                    else:
+                        background_left = 0
+
                     drawing_context.begin_path()
-                    drawing_context.move_to(0, baseline)
-                    drawing_context.line_to(0 + scale_marker_calculated_width, baseline)
-                    drawing_context.line_to(0 + scale_marker_calculated_width, baseline - scale_marker_height)
-                    drawing_context.line_to(0, baseline - scale_marker_height)
+                    drawing_context.rect(background_left - padding, background_top, content_width + padding * 2, total_height + padding)
                     drawing_context.close_path()
-                    drawing_context.fill_style = "#448"
+                    drawing_context.fill_style = "rgba(255, 255, 255, 0.6)"
                     drawing_context.fill()
-                    drawing_context.stroke_style = "#000"
-                    drawing_context.stroke()
+
+                # Draw scale bar
+                if self.__draw_on_right:
+                    bar_left = content_width - scale_marker_calculated_width
+                else:
+                    bar_left = 0
+
+                drawing_context.begin_path()
+                drawing_context.move_to(bar_left, baseline)
+                drawing_context.line_to(bar_left + scale_marker_calculated_width, baseline)
+                drawing_context.line_to(bar_left + scale_marker_calculated_width, baseline - scale_marker_height)
+                drawing_context.line_to(bar_left, baseline - scale_marker_height)
+                drawing_context.close_path()
+                drawing_context.fill_style = "#448"
+                drawing_context.fill()
+                drawing_context.stroke_style = "#000"
+                drawing_context.stroke()
+
+                # Draw text
+                if self.__include_info_text:
                     drawing_context.font = scale_marker_font
                     drawing_context.text_baseline = "bottom"
-                    drawing_context.fill_style = "#FFF"
-                    text1 = dimensional_calibration.convert_to_calibrated_size_str(scale_marker_image_width)
-                    text2 = info_text
-                    fm1 = get_font_metrics_fn(scale_marker_font, text1)
-                    drawing_context.fill_text(text1, 0, baseline - scale_marker_height - 4)
-                    drawing_context.fill_text(text2, 0, baseline - scale_marker_height - 4 - fm1.height)
+                if self.__draw_on_right:
```

Comment:
This sequence of `if`, `if`, `elif` is very misleading without a blank line before the 2nd `if`.

Also, the way this is written changes the drawing of the 2nd line based on whether the first line is drawn or not since the `font` and `baseline` are only set if `__include_info_text` is enabled.

Generally, I'd like to convert code like this to use the layout engine rather than doing the layout explicitly. But that can wait to another PR.

Repeating myself here: I'd like to eliminate `include_info_text` and address that in a future PR.